test/OBSSStorage.KetlKred.test.ts

```diff
@@ -0,0 +1,114 @@
+      {
+        inputs: [
+          { internalType: 'address', name: 'account', type: 'address' },
+          { internalType: 'uint256', name: 'id', type: 'uint256' },
+        ],
+        name: 'balanceOf',
+        outputs: [{ internalType: 'uint256', name: '', type: 'uint256' }],
+        stateMutability: 'view',
+        type: 'function',
+      },
+      {
+        inputs: [],
+        name: 'currentTokenId',
+        outputs: [{ internalType: 'uint32', name: '', type: 'uint32' }],
+        stateMutability: 'view',
+        type: 'function',
+      },
+    ])
+    ketlAttestationContractMock.balanceOf.returns(1)
+    ketlAttestationContractMock.currentTokenId.returns(1)
+
+    const profilesContractFactory = await ethers.getContractFactory('Profiles')
+    const profilesContract = (await upgrades.deployProxy(
+      profilesContractFactory,
+      [ketlAttestationContractMock.address, 0, deployer.address]
+    )) as Profiles
+    await profilesContract.deployed()
+
+    const ketlCredContractFactory = await ethers.getContractFactory('KetlCred')
+    const ketlCredContract = (await upgrades.deployProxy(
+      ketlCredContractFactory,
+      ['Ketl', 'KETL', 0, deployer.address],
+      {
+        initializer: 'initializeKetlCred',
+      }
+    )) as KetlCred
+    await ketlCredContract.deployed()
+
+    const feedsContractFactory = await ethers.getContractFactory('Feeds')
+    const feedsContract = (await upgrades.deployProxy(feedsContractFactory, [
+      ketlAttestationContractMock.address,
+      0,
+      deployer.address,
+    ])) as Feeds
+    await feedsContract.deployed()
+
+    const obssStorageFactory = await ethers.getContractFactory('OBSSStorage')
+    const obssStorageContract = (await upgrades.deployProxy(
+      obssStorageFactory,
+      [
+        zeroAddress,
+        version,
+        ketlCredContract.address,
+        profilesContract.address,
+        feedsContract.address,
+      ],
+      {
+        initializer: 'initialize',
+      }
+    )) as OBSSStorage
+    await obssStorageContract.deployed()
+
+    await ketlCredContract.setAllowedCaller(obssStorageContract.address)
+    await profilesContract.setAllowedCaller(obssStorageContract.address)
+    await feedsContract.setAllowedCaller(obssStorageContract.address)
+
+    return {
+      deployer,
+      user1,
+      user2,
+      user3,
+      profilesContract,
+      ketlCredContract,
+      feedsContract,
+      obssStorageContract,
+    }
+  }
+
+  it('should grant KetlCred to author when post is upvoted', async function () {
+    const {
+      user1,
+      deployer,
+      feedsContract,
+      obssStorageContract,
+      ketlCredContract,
+    } = await loadFixture(setupOBSS)
+    await feedsContract.addFeed(MOCK_CID)
+    expect(await feedsContract.lastFeedId()).to.equal(1)
+    await obssStorageContract.addFeedPost({
+      feedId: 0,
+      postMetadata: MOCK_CID,
+    })
+    await obssStorageContract.connect(user1).addFeedReaction({
```

Comment:
Let's also test switching reaction from from 1 to 2 (or from 0 to 1, can't quite remember):
User has `reactionType = 1`, ketl sends `addFeedReaction({ reactionType: 2 })`, it shouldn't throw an error, it should switch reactions from 1 to 2

Reply:
I'll definitely build out a full suite of tests for thsi!